routeros/resource_interface_ethernet.go

```diff
@@ -56,10 +56,10 @@ func ResourceInterfaceEthernet() *schema.Resource {
 			`"factory_name","driver_rx_byte","driver_rx_packet","driver_tx_byte","driver_tx_packet",` +
```

Comment:
Should we create a function `PropSkipFieldsSlice(items ...string)` So we can actually be sure that this not happen in the future? 
What do you guys think? @vaerh ?

Reply:
I mean something like: https://github.com/terraform-routeros/terraform-provider-routeros/pull/382

Reply:
@jlpedrosa I would rather change the existing `PropSkipFields` and replace all the occurrences to pass an array instead of a string. I also wanted to do that a while ago.

Reply:
I think it makes sense. But it should be done for all meta-attributes at once.
I changed the function a bit to make it work a bit faster:
``` go
func BenchmarkSlice1(b *testing.B) {
	for i := 0; i < b.N; i++ {
		PropSkipFieldsSlice1("0", "1","2", "3","4", "5","6", "7","8","9")
	}
}

func BenchmarkSlice2(b *testing.B) {
	for i := 0; i < b.N; i++ {
		PropSkipFieldsSlice2("0", "1","2", "3","4", "5","6", "7","8","9")
	}
}

func PropSkipFieldsSlice1(s ...string) string {
	builder := strings.Builder{}
	for _, item := range s {
		builder.WriteString(fmt.Sprintf("\"%s\",", item))
	}
	res := builder.String()
	lastComaRemoved := res[0 : len(res)-1]
	return lastComaRemoved
}

func PropSkipFieldsSlice2(s ...string) string {
	res := make([]string, len(s))
	for i, item := range s {
		res[i] = `"`+item+`"`
	}
	return strings.Join(res,",")
}
```
``` bash
BenchmarkSlice1
BenchmarkSlice1-8         671860              1533 ns/op             320 B/op         24 allocs/op
BenchmarkSlice2
BenchmarkSlice2-8        1836508               611.9 ns/op           240 B/op         12 allocs/op
```

``` go
func stringFromSlice(s []string) string {
	res := make([]string, len(s))
	for i, item := range s {
		res[i] = `"`+item+`"`
	}
	return strings.Join(res,",")
}

func PropSkipFields(s ...string) *schema.Schema {
	return &schema.Schema{
		Type:        schema.TypeString,
		Optional:    true,
		Default:     stringFromSlice(s),
		Description: "<em>A set of transformations for field names. This is an internal service field, setting a value is not required.</em>",
		DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
			return true
		},
	}
}
```

Reply:
I suggest we split this PR. We accept this one as it is.
Now I would like to start the process of publishing relises in the Hashi registry. There are problems with this.

Reply:
Actually, most likely the "heaviness" of my implementation comes from the sprintf, probably if you copy my loop with your string concat is going to be faster . Strings.join loops over the slice again. 100% up to you I don't think this is very performance sensitive.

Reply:
You're right. It's Sprintf
``` bash
BenchmarkSlice1
BenchmarkSlice1-8 2408424 477.1 ns/op 120 B/op 4 allocs/op
BenchmarkSlice2
BenchmarkSlice2-8 1857993 616.3 ns/op 240 B/op 12 allocs/op
```

Reply:
@vaerh Could we maybe merge this PR?   It should not be a problem to release it later.